widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java

```diff
@@ -420,18 +422,25 @@ public void onMessageTextChanged(String message) {
                         .setLastTypedText(message)
                         .setShowSendButton(isShowSendButtonUseCase.execute(message))
         );
-        if (chatState.isOperatorOnline()) {
-            Logger.d(TAG, "Send preview: " + message);
-            sendMessagePreviewUseCase.execute(message);
-        }
+        sendMessagePreview(message);
     }
 
     public void sendMessage(String message) {
         Logger.d(TAG, "Send MESSAGE: " + message);
-        Logger.d(TAG, "SEND MESSAGE sending");
+        clearMessagePreview();
         sendMessageUseCase.execute(message, sendMessageCallback);
     }
 
+    private void sendMessagePreview(String message) {
+        if (chatState.isOperatorOnline()) {
+            sendMessagePreviewUseCase.execute(message);
+        }
+    }
+
+    private void clearMessagePreview() {
+        sendMessagePreview(EMPTY_MESSAGE);
+    }
```

Comment:
Thats how the Core SDK works. Empty string has to be sent to clear the "visitor is typing" status in the hub. I created a separate method to make this more clear. Otherwise one may ask "Why do you even need to send the empty message". Maybe we should hide that peculiar Core SDK feature in the repository and just call `clearMessagePreview()` in the controller?

Reply:
I could remove that method and add a comment to clarify why empty message is sent. Would that be better?

Reply:
Ah, I see now. I would leave it in a separate function then, and just add a comment to line #441 that an empty string has to be sent to clear the "visitor is typing" status in the hub.

Reply:
> I could remove that method and add a comment to clarify why empty message is sent. Would that be better?

This idea is good as well.